opm/simulators/wells/RateConverter.hpp

```diff
@@ -496,20 +496,34 @@ namespace Opm {
                     if (hydrocarbonPV > 0.) {
                         auto& attr = attributes_hpv[reg];
                         attr.pv += hydrocarbonPV;
-                        attr.pressure += fs.pressure(FluidSystem::oilPhaseIdx).value() * hydrocarbonPV;
-                        attr.rs += fs.Rs().value() * hydrocarbonPV;
-                        attr.rv += fs.Rv().value() * hydrocarbonPV;
-                        attr.temperature += fs.temperature(FluidSystem::oilPhaseIdx).value() * hydrocarbonPV;
+                        if (Details::PhaseUsed::oil(pu)) {
+                            attr.pressure += fs.pressure(FluidSystem::oilPhaseIdx).value() * hydrocarbonPV;
+                            attr.rs += fs.Rs().value() * hydrocarbonPV;
+                            attr.rv += fs.Rv().value() * hydrocarbonPV;
+                            attr.temperature += fs.temperature(FluidSystem::oilPhaseIdx).value() * hydrocarbonPV;
+                        }
+                        else{
+                            assert(Details::PhaseUsed::gas(pu));
+                            attr.pressure += fs.pressure(FluidSystem::gasPhaseIdx).value() * hydrocarbonPV;
+                            attr.temperature += fs.temperature(FluidSystem::gasPhaseIdx).value() * hydrocarbonPV;
+                        }
                         attr.saltConcentration += fs.saltConcentration().value() * hydrocarbonPV;
                     }
 
                     if (pv_cell > 0.) {
                         auto& attr = attributes_pv[reg];
                         attr.pv += pv_cell;
-                        attr.pressure += fs.pressure(FluidSystem::oilPhaseIdx).value() * pv_cell;
-                        attr.rs += fs.Rs().value() * pv_cell;
-                        attr.rv += fs.Rv().value() * pv_cell;
-                        attr.temperature += fs.temperature(FluidSystem::oilPhaseIdx).value() * pv_cell;
+                        if (Details::PhaseUsed::oil(pu)) {
+                            attr.pressure += fs.pressure(FluidSystem::oilPhaseIdx).value() * pv_cell;
+                            attr.rs += fs.Rs().value() * pv_cell;
+                            attr.rv += fs.Rv().value() * pv_cell;
+                            attr.temperature += fs.temperature(FluidSystem::oilPhaseIdx).value() * pv_cell;
+                        }
+                        else{
```

Comment:
very nitpicking here, please do `} else {` to make it consistent with other code in the same file.